bindings/php/src/lib.rs

```diff
@@ -15,14 +15,165 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::collections::HashMap;
+use std::str::FromStr;
 use ext_php_rs::prelude::*;
-use opendal::{EntryMode, Metadata};
+use ext_php_rs::{exception::PhpException, zend::ce};
+use ext_php_rs::convert::FromZval;
+use ext_php_rs::flags::DataType;
+use ext_php_rs::types::Zval;
+use ::opendal as od;
 
-#[php_function]
-pub fn debug() -> String {
-    let metadata = Metadata::new(EntryMode::FILE);
+#[php_class(name = "OpenDAL\\Operator")]
+pub struct Operator(od::BlockingOperator);
 
-    format!("{:?}", metadata)
+#[php_impl]
+impl Operator {
+    pub fn __construct(scheme_str: String, mp: HashMap<String, String>) -> PhpResult<Self> {
+        let scheme = od::Scheme::from_str(&scheme_str).map_err(format_php_err)?;
+        let op = od::Operator::via_map(scheme, mp).map_err(format_php_err)?;
+
+        Ok(Operator(op.blocking()))
+    }
+
+    /// Write bytes into given path.
+    pub fn write(&self, path: &str, content: String) -> PhpResult<()> {
```

Comment:
Does php has concept lile `Bytes` or `Vec<u8>`? It's incorrect to assume that all content are `UTF-8` string.

Reply:
PHP does not support byte arrays. Usually, we use strings directly in PHP. However, PHP supports streams. I will investigate how to support streams in the next PR.

Reply:
Maybe we need to use https://davidcole1340.github.io/ext-php-rs/types/binary.html?

Reply:
Yes, I saw that too. However, if we use Binary here, we need to use pack in PHP to convert data into binary data. This is not common in PHP, so in my opinion we can directly use string here.

```rust
#[php_function]
pub fn test_binary(input: Binary<u32>) -> Binary<u32> {
}
```


```php
$data = pack('*L', [1, 2, 3, 4, 5]);
$output = unpack('*L', test_binary($data));
```

Reply:
> so in my opinion we can directly use string here.

No, we can't. Mapping PHP `String` (which could be invalid UTF-8) to Rust `String` (which must be UTF-8) is wrong and could be unsound. 

How about raising an issue upstream to seek fo some advice?

Reply:
I added a new test that throws an exception if the UTF-8 encoding is invalid during conversion in ext-php-rs.

```php
it('invalid UTF-8 encoding', function () {
    $op = new \OpenDAL\Operator('fs', ['root' => '/tmp']);

    $op->write('test.txt', 'invalid UTF-8: '.chr(0x80));
})->throws('Exception')->expectExceptionMessage('Invalid value given for argument `content`.');
```

Reply:
Also, added two binary-safe read/write methods, please help take a look.